ceph/ceph_admin/osd.py

```diff
@@ -19,34 +19,31 @@ class DevicesNotFound(Exception):
 
 
 class OSD(ApplyMixin, Orch):
+
     """Interface to ceph orch osd."""
 
     SERVICE_NAME = "osd"
 
-    def device_list(self, refresh=False):
+    def apply(self, config: Dict) -> None:
         """
-        Return all available devices using "orch device ls" command.
+        Deploy the ODS service using the provided configuration.
 
         Args:
-            refresh:    Perform refresh before retrieving the device list.
-
-        Returns:
-            device_list: list of nodes with available devices
+            config: Key/value pairs provided by the test case to create the service.
 
-        device_list:
-            node1: ["/dev/sda", "/dev/sdb"]
-            node2: ["/dev/sda"]
+        Example
+            config:
+                command: apply
+                service: osd
+                base_cmd_args:          # arguments to ceph orch
+                    concise: true
+                    verbose: true
+                args:
+                    all-available-devices: true
+                    dry-run: true
+                    unmanaged: true
```

Comment:
I think `refresh` code part is pretty much needed, just to make sure all devices are available.
please add back that code.

Reply:
Found that its not called in `device ls` definition as well.